src/components/Stake.svelte

```diff
@@ -0,0 +1,163 @@
+<script lang="ts">
+  import {
+    dotApi,
+    storeConnected,
+    storeCurrentAction,
+    storeProviderId,
+    storeToken,
+    storeValidAccounts,
+    transactionSigningAddress,
+  } from '$lib/stores';
+  import type { ApiPromise } from '@polkadot/api';
+  import { DOLLARS, submitStake } from '$lib/connections';
+  import { ActionForms, defaultDotApi } from '$lib/storeTypes';
+  import KeySelection from './KeySelection.svelte';
+  import { onMount } from 'svelte';
+  import { isFunction } from '@polkadot/util';
+  import { isLocalhost } from '$lib/utils';
+  import TransactionStatus from './TransactionStatus.svelte';
+  import type { web3Enable, web3FromSource } from '@polkadot/extension-dapp';
+  import type { AccountMap } from '$lib/polkadotApi';
+
+  let connected = false;
+  let thisDotApi = defaultDotApi;
+  let signingAddress: string = ''; // eslint-disable-line no-unused-vars
+  let showSelf: boolean = false; // eslint-disable-line no-unused-vars
+  let selectedKey: string = '';
+  let thisWeb3FromSource: typeof web3FromSource;
+  let thisWeb3Enable: typeof web3Enable;
+  let showTransactionStatus = false;
+  let stakeAmount: bigint = 1n;
+  let token = '';
+  export let txnStatuses: Array<string> = [];
+
+  $: stakeAmountInDollars = BigInt.asUintN(64, stakeAmount) * BigInt.asUintN(64, DOLLARS);
+
+  onMount(async () => {
+    const extension = await import('@polkadot/extension-dapp');
+    thisWeb3FromSource = extension.web3FromSource;
+    thisWeb3Enable = extension.web3Enable;
+  });
+
+  export let providerId = 0;
+  export let validAccounts: AccountMap = {};
+
+  storeConnected.subscribe((val) => (connected = val));
+  dotApi.subscribe((api) => (thisDotApi = api));
+  transactionSigningAddress.subscribe((val) => (signingAddress = val));
+  storeCurrentAction.subscribe((val) => (showSelf = val == ActionForms.Stake));
+  storeProviderId.subscribe((val) => (providerId = val));
+  storeValidAccounts.subscribe((val) => (validAccounts = val));
+  storeToken.subscribe((val) => (token = val));
+
+  const hideSelf = () => {
+    storeCurrentAction.update((val) => (val = ActionForms.NoForm));
+    clearTxnStatuses();
+  };
+
+  const addNewTxnStatus = (txnStatus: string) => {
+    txnStatuses = [...txnStatuses, txnStatus];
+  };
+  const clearTxnStatuses = () => (txnStatuses = new Array<string>());
+
+  const stake = async (evt: Event) => {
+    evt.preventDefault();
+    clearTxnStatuses();
+    let endpointURI: string = thisDotApi.selectedEndpoint || '';
+    if (selectedKey === '') {
+      alert('Please choose a key to stake from.');
+    } else if (isFunction(thisWeb3FromSource) && isFunction(thisWeb3Enable)) {
+      let signingKeys = validAccounts[selectedKey];
+      showTransactionStatus = true;
+      if (isLocalhost(endpointURI)) {
```

Comment:
Non blocking:  I realized we're actually doing the web3Enable and web3FromSource checks before checking for localhost, when we don't need to.  In my PR I move these checks inside the `else` below for when we use an extension.

Reply:
Moved.